File: go/chessstrategy/chess_strategy.go
package chessstrategy

type ChessStrategy struct { Turn uint64; Resources map[string]int64 }
func New() *ChessStrategy { return &ChessStrategy{Resources: map[string]int64{"gold":100,"food":100,"science":0}} }
func (s *ChessStrategy) Tick(resource string, amount int64) uint64 { s.Resources[resource] += amount; s.Turn++; return s.Turn }
type UCIRequest struct { Protocol string `json:"protocol"`; FEN string `json:"fen"`; Command string `json:"command"` }
func (s *ChessStrategy) UCI(fen, command string) UCIRequest { if command=="" { command="go movetime 100" }; return UCIRequest{"UCI",fen,command} }

File: mobile/kotlin/src/main/kotlin/bizx/chess/ChessStrategy.kt
package bizx.chess

data class StrategyState(var turn: Long = 0, val resources: MutableMap<String, Long> = mutableMapOf("gold" to 100, "food" to 100, "science" to 0))

class ChessStrategy {
    val state = StrategyState()
    fun tick(resource: String, amount: Long) = state.resources.merge(resource, amount, Long::plus).let { ++state.turn }
    fun uci(fen: String, command: String = "go movetime 100") = mapOf("protocol" to "UCI", "fen" to fen, "command" to command, "adapter" to "external-engine")
}
